src/main/scala/cromwell/engine/db/slick/ExecutionComponent.scala

```diff
@@ -68,10 +74,23 @@ trait ExecutionComponent {
     } yield execution.status)
 
   // see workflowExecutionsByStatuses
-  def executionStatusesByWorkflowExecutionIdAndCallFqns(workflowExecutionId: Int, callFqns: Traversable[String]) = {
-    for {
+  def executionStatusesByWorkflowExecutionIdAndCallFqns(workflowExecutionId: Int, scopeKeys: Traversable[ExecutionDatabaseKey]) = {
+    val scopeFqns = scopeKeys.map(_.fqn).toIterable
+    val scopeIndexes = scopeKeys.map( k => ExecutionIndex.fromIndex(k.index)).toIterable
+    val scopeID = scopeFqns zip scopeIndexes
+    /*
+     * FIXME: This is bad, there is probably a better way
+     * We want entries that have the right workflowID AND a (fqn, index) pair which is in scopeKeys
+     * Use workaround because slick does not supporting tuples with inSet ATM: https://github.com/slick/slick/pull/995 
+     */
+    val falseRep: Rep[Boolean] = false
+    (for {
       execution <- executions
-      if execution.workflowExecutionId === workflowExecutionId && (execution.callFqn inSet callFqns)
-    } yield execution.status
+      if execution.workflowExecutionId === workflowExecutionId
+    } yield execution) filter { exec =>
```

Comment:
I don't really like wrapping for comps in parens/braces just so one can chain stuff after it. Can you assign the for comp to a val and then operate on that?